src/Microsoft.Management.Configuration.Processor/Helpers/PowerShellHelpers.cs

```diff
@@ -48,24 +48,41 @@ internal static class PowerShellHelpers
 
             if (!string.IsNullOrEmpty(version))
             {
+                // Prerelease versions append the prerelease tag after a -
+                // PowerShell doesn't handle semantic versions.
+                if (version.Contains("-"))

```

Comment:
Maybe a function? `GetOnlyNumericVersion` or something?  Or a new type that splits them out.  It could be used to implicitly declare `allowPrerelease` by having something after the `-`.